### PR TITLE
Prone stance with bipod utilization

### PR DESCRIPTION
**Pull request by Rainfey**
#### Summary
SUMMARY: Features "Adds the ability for the player to go prone and use bipods"

#### Purpose of change
Currently the bipod benefits are only available on MOUNTABLE terrain, where it should also be possible to utilize one whilst prone provided the surface is flat.

#### Describe the solution
* Additional functionality has been added to lines that deal with crouching to handle the new prone stance
* During the bipod check an additional check for being in the prone stance on terrain marked as FLAT has been added
* Prone reduces the visibility of a player by a factor of 3 compared to crouching (possible to be hidden completely by a coverage of 20 vs 60 for crouching)
* Prone allows vision to be blocked by a coverage of >=10
* Prone movement has a cost of 2x that of crouching (~400 vs 200 for crouched)
* Medium length grass tiles have been modified to have a coverage of 15, more tiles may need to be modified as appropriate
* Toggling prone (via the 'Toggle Prone' keybind) results in the character switching between a prone and crouched stance rather than prone and standing

#### Describe alternatives you've considered
* A standing/telescopic bipod that can be deployed whilst standing
* Allowing the existing bipod to work on flat terrain whilst crouched

#### Testing
* Tested that the player is capable of switching between stances correctly
* Tested hiding behind medium length grass, including vision blocking
* Tested bipod usage whilst prone (debug checks indicate its working but the ingame effect was not able to be conclusively confirmed)
* Tested changing stance whilst mounted on a horse (prone stance is blocked whilst mounted)

#### Additional context
None

### CODE REVIEW

File: src/handle_action.cpp [hunk 1]
```diff
@@ -1354,6 +1354,7 @@ static void open_movement_mode_menu()
     as_m.entries.emplace_back( 0, true, 'w', _( "Walk" ) );
     as_m.entries.emplace_back( 1, true, 'r', _( "Run" ) );
     as_m.entries.emplace_back( 2, true, 'c', _( "Crouch" ) );
+    as_m.entries.emplace_back( 3, true, 'c', _( "Prone" ) );
```
Comment (anothersimulacrum) [inline src/handle_action.cpp]:
```suggestion
    as_m.entries.emplace_back( 3, true, 'p', _( "Prone" ) );
```

**Comment by dissociativity** [comment]
I would increase the move cost to 3x crouching, moving while prone is by far more difficult and tiring than moving while crouching no matter how fit you are.

**Comment by ScampsAdams** [comment]
Is there move cost to stand up? Seems unbalanced otherwise, free precision.

**Comment by Rainfey** [comment]
> 
> 
> I would increase the move cost to 3x crouching, moving while prone is by far more difficult and tiring than moving while crouching no matter how fit you are.

I'm open to changing the movement cost as 2x was based on scattered sources of human locomotion speeds, primarily on the table available at https://deadbydaylight.gamepedia.com/Movement_Speeds which is not exactly definitive.

>
>
> Is there move cost to stand up? Seems unbalanced otherwise, free precision.

Does anyone have any suggestions on how to accomplish this?

**Comment by Brian-Otten** [comment]
Movecost in going prone has been the main obstacle in the past because if going prone has a movecost, you can't have prone activated using the same "cycle" through the stances button we currently use for run/walk/crouch, because it would get you killed when you cycle through prone, so it would require it's own button/hotkey.

So you'll need to find a way to have this cost some movepoints, and a good keybind to activate it before this will be ready to merge.
I do hope you can make it work because this has been on my wishlist for a while.

**Comment by Rainfey** [comment]
Regarding the move cost for going prone, the easiest method of acomplishing this is to change the character_movemode enum to the following : 

```
enum character_movemode : unsigned char {
    CMM_WALK = 0,
    CMM_RUN = 1,
    CMM_CROUCH = 2,
    CMM_COUNT,

    //Done to exclude the prone stance from the stance cycle keybind
    CMM_PRONE,
};
```
This appears to work without issue, however it feels like the kind of solution that will result in unforseen issues.
I would like to solicit advice on this before committing.

**Comment by Rainfey** [comment]
I'm closing this in favor of a more substantial rewrite of the movement mode system

**Pull request closed without merging**